vllm/v1/engine/async_llm.py

```diff
@@ -140,20 +134,33 @@ def _send_to_streams(self, request_outputs: List[RequestOutput]):
 
         for request_output in request_outputs:
             request_id = request_output.request_id
-            assert request_id in self.request_streams
 
-            self.request_streams[request_id].put(request_output)
+            stream = self.request_streams.get(request_id)
+            if stream is not None:
+                finished = request_output.finished
+                stream.put(request_output)
+                if finished:
+                    self._finish_stream(request_id)
 
-            if request_output.finished:
-                self.request_streams[request_id].finish()
-                self.request_streams.pop(request_id)
+    def _finish_stream(self, request_id: str):
+        stream = self.request_streams.pop(request_id)
+        if stream is not None:
+            stream.finish()
```

Comment:
We need to remove the stream if it finishes due to a finished request output or due to client cancellation.